ash/src/extensions/khr/display.rs

```diff
@@ -0,0 +1,203 @@
+use crate::prelude::*;
+use crate::version::{EntryV1_0, InstanceV1_0};
+use crate::vk;
+use crate::RawPtr;
+use std::ffi::CStr;
+use std::mem;
+use std::ptr;
+
+#[derive(Clone)]
+pub struct Display {
+    handle: vk::Instance,
+    display_fn: vk::KhrDisplayFn,
+}
+
+impl Display {
+    pub fn new<E: EntryV1_0, I: InstanceV1_0>(entry: &E, instance: &I) -> Display {
+        let display_fn = vk::KhrDisplayFn::load(|name| unsafe {
+            mem::transmute(entry.get_instance_proc_addr(instance.handle(), name.as_ptr()))
+        });
+        Display {
+            handle: instance.handle(),
+            display_fn,
+        }
+    }
+
+    pub fn name() -> &'static CStr {
+        vk::KhrDisplayFn::name()
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/vkGetPhysicalDeviceDisplayPropertiesKHR.html>"]
+    pub unsafe fn get_physical_device_display_properties(
+        &self,
+        physical_device: vk::PhysicalDevice,
+    ) -> VkResult<Vec<vk::DisplayPropertiesKHR>> {
+        let mut count = 0;
+        self.display_fn.get_physical_device_display_properties_khr(
+            physical_device,
+            &mut count,
```

Comment:
Can you elaborate on why this might be necessary? That change introduces a new operator; I don't see anything about deprecating implicit conversion of references to pointers.

Reply:
I guess it doesn't matter in this case since we can reliably create a reference. In situations where we cannot reliably create a reference (that is, we cannot prove that the data is aligned, non-null, and has a well-defined lifetime), we should avoid using `&mut x` since that can trigger undefined behaviour. An example of a situation which would be problematic is `mem::uninitialized()`. `&mut x` where `x` is uninitialized is undefined behaviour, even if you never dereference, but `&raw mut x` or `&mut x as *mut _` is well defined behaviour so as long as you don't dereference the pointer before initializing the memory. The syntax `&mut x as *mut _` is only a temporary hack and will be deprecated in favour of `&raw mut x`.

Reply:
> The syntax &mut x as *mut _ is only a temporary hack and will be deprecated in favour of &raw mut x

As far as I can tell, that's only true when `x` is not a real value. The conversions in this PR from valid references to raw pointers is idiomatic and should remain as-is.